source/text/implementation/vss-strings-utf8.adb

```diff
@@ -70,6 +70,30 @@ package body VSS.Strings.UTF8 is
    --  Destination. Length is set to the length of the text in characters.
    --  Success is set False when validation is failed and to True otherwise.
 
+   ----------------------------
+   -- Before_First_Character --
+   ----------------------------
+
+   overriding procedure Before_First_Character
+     (Self     : UTF8_String_Handler;
+      Data     : String_Data;
+      Position : in out VSS.Strings.Cursor) is
+   begin
+      Position := (Index => 0, UTF8_Offset => 0, UTF16_Offset => 0);
```

Comment:
Should it be `UTF8_Offset => VSS.Unicode.UTF8_Code_Unit_Index'Last`?

Reply:
What is rationale behind it?

Reply:
to match default initialization.

Reply:
Default initialization should create invalid cursor.